### ISSUE TITLE
Update paginate function to support the Gitlab API

### ISSUE THREAD

**Issue by mrsaicharan1** [chaoss/augur#742]
**Issue Description**
 The paginate function only works with the GitHub API as of now. Without breaking the previous functionality, I'll add support for GitLab API too. Diving deep into the issue, the str indexing doesn't return the last page when we try to access `r.links['last']['url'][-6:].split('=')[1]`

Just as a heads up, gitlab performs pagination in the same way as github(link headers)
https://github.com/chaoss/augur/blob/e8e886a36157c32a605b7e90950c266a88025ad9/workers/standard_methods.py#L308

For reference:
The link headers sent back from the Gitlab API are in the format mentioned below.
```
{   'first': {   'rel': 'first',
                 'url': 'https://gitlab.com/api/v4/projects/36189/issues?id=36189&order_by=created_at&page=1&per_page=100&scope=all&sort=desc&state=all&with_labels_details=false'},
    'last': {   'rel': 'last',
                'url': 'https://gitlab.com/api/v4/projects/36189/issues?id=36189&order_by=created_at&page=20&per_page=100&scope=all&sort=desc&state=all&with_labels_details=false'}}
```

**Comment by mrsaicharan1** [comment]
In addition the links header response, the sample response for one of the issues in a project
```
{   '_links': {   'award_emoji': 'https://gitlab.com/api/v4/projects/7173393/issues/47/award_emoji',
                      'notes': 'https://gitlab.com/api/v4/projects/7173393/issues/47/notes',
                      'project': 'https://gitlab.com/api/v4/projects/7173393',
                      'self': 'https://gitlab.com/api/v4/projects/7173393/issues/47'},
        'assignee': None,
        'assignees': [],
        'author': {   'avatar_url': 'https://secure.gravatar.com/avatar/0f45334db94b5537ecaa58e3abe2a8cb?s=80&d=identicon',
                      'id': 1385824,
                      'name': 'Tim Mayberry',
                      'state': 'active',
                      'username': 'mojofunk',
                      'web_url': 'https://gitlab.com/mojofunk'},
        'closed_at': '2020-05-29T23:09:27.526Z',
        'closed_by': {   'avatar_url': 'https://secure.gravatar.com/avatar/0f45334db94b5537ecaa58e3abe2a8cb?s=80&d=identicon',
                         'id': 1385824,
                         'name': 'Tim Mayberry',
                         'state': 'active',
                         'username': 'mojofunk',
                         'web_url': 'https://gitlab.com/mojofunk'},
        'confidential': False,
        'created_at': '2020-05-29T11:08:18.872Z',
        'description': 'As: A developer\n'
                       'I Want: moc generated cpp files to be compiled as a '
                       'single cpp source file\n'
                       'So That: builds are faster\n'
                       '\n'
                       'With the way qadt is currently using the Qt support in '
                       'CMake the moc generated source files are compiled a '
                       'single file at a time. There are 20+ moc_*cpp files '
                       'generated as part of the qadt library and a build may '
                       'be significantly faster if they are all compiled as a '
                       'single source file.\n'
                       '\n'
                       'See: https://youtu.be/MJ4DunSrMFw?t=507',
        'discussion_locked': None,
        'downvotes': 0,
        'due_date': None,
        'has_tasks': False,
        'id': 35166835,
        'iid': 47,
        'labels': [],
        'merge_requests_count': 0,
        'milestone': None,
        'moved_to_id': None,
        'project_id': 7173393,
        'references': {   'full': 'mojofunk/qadt#47',
                          'relative': '#47',
                          'short': '#47'},
        'state': 'closed',
        'task_completion_status': {'completed_count': 0, 'count': 0},
        'time_stats': {   'human_time_estimate': None,
                          'human_total_time_spent': None,
                          'time_estimate': 0,
                          'total_time_spent': 0},
        'title': 'Use CMAKE AUTOMOC to compile moc generated cpp files as a '
                 'single object, amalgamated source file',
        'updated_at': '2020-05-29T23:10:12.747Z',
        'upvotes': 0,
        'user_notes_count': 0,
        'web_url': 'https://gitlab.com/mojofunk/qadt/-/issues/47',
        'weight': None},

```

**Comment by mrsaicharan1** [comment]
PR merged - #743
